plugins/inputs/ds389/ds389.go

```diff
@@ -0,0 +1,373 @@
+	BindPassword       string
+	Dbtomonitor        []string
+	AllDbmonitor       bool
+	Status             bool
+}
+
+const sampleConfig string = `
+  host = "ldap_instance"
+  port = 389
+
+  # ldaps, starttls, or no encryption. default is an empty string, disabling all encryption.
+  # note that port will likely need to be changed to 636 for ldaps
+  # valid options: "" | "starttls" | "ldaps"
+  tls = ""
+
+  # skip peer certificate verification. Default is false.
+  insecure_skip_verify = false
+
+  # Path to PEM-encoded Root certificate to use to verify server certificate
+  tls_ca = "/etc/ssl/certs.pem"
+
+  # dn/password to bind with. If bind_dn is empty, an anonymous bind is performed.
+  bind_dn = ""
+  bind_password = ""
+
+  #Gather dbname monitor
+  # Comma separated list of db filename
+  # dbtomonitor = ["exampleDB"]
+  # If true, alldbmonitor monitors all db and it overrides "dbtomonitor".
+  alldbmonitor = false
+
+  # Connections status monitor
+  status = false
+`
+
+var searchMonitor = "cn=Monitor"
+var searchLdbmMonitor = "cn=monitor,cn=ldbm database,cn=plugins,cn=config"
+
+var searchFilter = "(objectClass=extensibleObject)"
+var searchAttrs = []string{
+	"currentconnections",
+	"totalconnections",
+	"currentconnectionsatmaxthreads",
+	"maxthreadsperconnhits",
+	"dtablesize",
+	"readwaiters",
+	"opsinitiated",
+	"opscompleted",
+	"entriessent",
+	"bytessent",
+	"anonymousbinds",
+	"unauthbinds",
+	"simpleauthbinds",
+	"strongauthbinds",
+	"bindsecurityerrors",
+	"inops",
+	"readops",
+	"compareops",
+	"addentryops",
+	"removeentryops",
+	"modifyentryops",
+	"modifyrdnops",
+	"listops",
+	"searchops",
+	"onelevelsearchops",
+	"wholesubtreesearchops",
+	"referrals",
+	"chainings",
+	"securityerrors",
+	"errors",
+	"connections",
+	"connectionseq",
+	"connectionsinmaxthreads",
+	"connectionsmaxthreadscount",
+	"bytesrecv",
+	"bytessent",
+	"entriesreturned",
+	"referralsreturned",
+	"masterentries",
+	"copyentries",
+	"cacheentries",
+	"cachehits",
+	"slavehits",
+	"backendmonitordn",
+	"connection",
+	"version",
+}
+
+var searchLdbmAttrs = []string{
+	"dbcachehitratio",
+	"dbcachehits",
+	"dbcachepagein",
+	"dbcachepageout",
+	"dbcacheroevict",
+	"dbcacherwevict",
+	"dbcachetries",
+}
+
+var searchDbAttrs = []string{}
+
+func (o *ds389) SampleConfig() string {
+	return sampleConfig
+}
+
+func (o *ds389) Description() string {
+	return "ds389 cn=Monitor plugin"
+}
+
+// return an initialized ds389
+func Newds389() *ds389 {
+	return &ds389{
+		Host:               "localhost",
+		Port:               389,
+		SSL:                "",
+		TLS:                "",
+		InsecureSkipVerify: false,
+		SSLCA:              "",
+		TLSCA:              "",
+		BindDn:             "",
+		BindPassword:       "",
+		Dbtomonitor:        []string{},
+		AllDbmonitor:       false,
+		Status:             false,
+	}
+}
+
+// gather metrics
+func (o *ds389) Gather(acc telegraf.Accumulator) error {
+	if o.TLS == "" {
+		o.TLS = o.SSL
+	}
+	if o.TLSCA == "" {
```

Comment:
Using internal/tls.ClientConfig will allow this to go away, also simplifying/affecting line 164